### PR TITLE
docs: add composables docs

### PR DESCRIPTION
**Pull request by filipsobol**
This PR explains what composables are (with Magento-specific examples) and creates one page that lists all composables instead of listing each composable in the sidebar.

Preview:
localhost_8080_composition_composables html

### CODE REVIEW

**Review by Frodigo — APPROVED** [review]
Thanks for that @filipsobol, it's awesome! 

Please resolve conflicts.

**Comment by filipsobol** [comment]
@Frodigo I resolved the conflict. I also asked @bartoszherba to review certain parts of these docs that we discussed

File: docs/composition/composables.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    };
+  }
+};
+</script>
+```
+
+Let's go step by step through this example to understand what's going on:
+
+1. We begin by extracting needed methods and state variables from the composable.
+2. Next, we call the asynchronous `load` method within the `useFetch` lifecycle hook to load user data.
```
Comment (bartoszherba) [inline docs/composition/composables.md:66]:
```suggestion
2. Next, we call the asynchronous `load` method within the `useFetch` hook to load user data.
```

File: docs/composition/composables.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+const { user } = useUser();
+const firstname = computed(() => user.value.firstname);
+```
+
+This raises two questions:
+
+1. What is the `useFetch`, and what does it do?
+2. What happened when we called the `load` method?
+
+### `useFetch` and other lifecycle hooks for data-fetching
```
Comment (bartoszherba) [inline docs/composition/composables.md:85]: Hey, useFetch is just a hook, not a lifecycle hook, lifecycle hooks -> https://vuejs.org/api/composition-api-lifecycle.html
useFetch though is executed with the usage of lifecycle hooks internally, a few of them - so it spans over multiple phases.

File: docs/composition/composables.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+1. It set the `loading` property to `true`.
+2. It called the corresponding API endpoint to load the current user's data.
+   - if the request succeeded, it updated the `user` property,
+   - otherwise, it added the error message to the `error` property.
+3. It set the `loading` property to `false`.
+
+## What's next?
+
+Now that you understand what composables are, you should see the [List of composables](./list-of-composables.html) and their options.
```
Comment (bartoszherba) [inline docs/composition/composables.md:114]: In the list of composables, external checkout composable is not finished yet.

File: docs/composition/list-of-composables.md [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+
+See the [UseCountrySearchInterface](/api-reference/magento-theme.usecountrysearchinterface.html) for more information.
+
+## `useCurrency`
+
+Allows loading and changing the currency.
+
+See the [UseCurrencyInterface](/api-reference/magento-theme.usecurrencyinterface.html) for more information.
+
+## `useExternalCheckout`
```
Comment (bartoszherba) [inline docs/composition/list-of-composables.md:63]: Can we mark this one as unfinished?

**Review by bartoszherba — CHANGES_REQUESTED** [review]
All is good but useFetch description must be changes to avoid naming this hook as a lifecycle hook, please check https://composition-api.nuxtjs.org/lifecycle/usefetch/

File: docs/composition/list-of-composables.md [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+
+Allows redirecting to external checkout process. It depends on the [magento2-external-checkout repository](https://github.com/Vendic/magento2-external-checkout).
+
+See the [UseExternalCheckoutInterface](/api-reference/magento-theme.useexternalcheckoutinterface.html) for more information.
+
+## `useFacet`
+
+Allows searching for products using facets. It accepts multiple filters, allowing to narrow down the results to a specific category, search term, etc.
```
Comment (bartoszherba) [inline docs/composition/list-of-composables.md:71]: I am not sure about this, this is basically `products` query but the response is extended by pagination, totals, and sorting options, so it is not about searching with the usage of something like `facet` but to have a response enriched with additional data, where some are not even fetched from the API.

File: docs/composition/list-of-composables.md [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+
+## `useUiHelpers`
+
+Allows handling the parameters for filtering, searching, sorting and pagination in the URL search/query params.
+
+See the [UseUiHelpersInterface](/api-reference/magento-theme.useuihelpersinterface.html) for more information.
+
+## `useUiNotification`
+
+Allows showing notifications to the user.
```
Comment (bartoszherba) [inline docs/composition/list-of-composables.md:167]: More like helps manage notifications? I mean, showing notifications is just a part of its functions.

**Review by bartoszherba — APPROVED** [review]

**Pull request merged**